src/components/ResourceBrowser/ResourceList/K8SResourceList.tsx

```diff
@@ -139,6 +147,31 @@ export const K8SResourceList = ({
 
     const showPaginatedView = resourceList?.data?.length >= pageSize
 
+    /**
+     * Initial Sort Key
+     *
+     * This constant holds the value for the initial sort key. \
+     * Since our headers are dynamic and the default key 'namespace' might or might not be present,
+     * we switch to the key 'name', which will always be present.
+     */
+    const initialSortKey = useMemo(() => {
+        if (resourceList) {
+            const isNameSpaceColumnPresent = resourceList.headers.findIndex((header) => header === 'namespace')
+            return isNameSpaceColumnPresent < 0 ? 'name' : 'namespace'
+        }
+        return ''
+    }, [resourceList])
+
+    // SORTING HOOK
+    const { sortBy, sortOrder, handleSorting, clearFilters } = useStateFilters({ initialSortKey })
+
+    useEffect(() => {
+        /** Reset the sort filters when initial sort key is updated after api response. */
+        if (initialSortKey) {
+            clearFilters()
```

Comment:
are you sure we want to clear all the filters here?

Reply:
yes, we are not using search via useStateFilters